--- configure.ac.orig	2014-10-03 10:32:41.900677372 +0000
+++ configure.ac	2014-10-03 10:32:50.364796760 +0000
@@ -82,6 +82,12 @@
   esac],[with_xenstore=true]) 
 AM_CONDITIONAL(WITH_XENSTORE, test x$with_xenstore = xtrue)
 
+dnl The dlopen() function is in the C library for *BSD and in
+dnl libdl on GLIBC-based systems
+AC_SEARCH_LIBS([dlopen], [dl dld], [], [
+  AC_MSG_ERROR([unable to find the dlopen() function])
+])
+
 AC_OUTPUT(vhostmd/Makefile
           include/Makefile
           libmetrics/Makefile
